### PR TITLE
testing for RunCommitRuleOnBestChain

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
Current dependencies on/for this PR:
* feature/pos-fast-hotstuff-consensus
  * **PR #742** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/742" target="_blank">Graphite</a> 
    * **PR #746** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/746" target="_blank">Graphite</a> 
      * **PR #747** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/747" target="_blank">Graphite</a>  👈
        * **PR #748** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/748" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/747?utm_source=stack-comment).

File: lib/pos_blockchain.go [hunk 1]
```diff
@@ -530,6 +533,17 @@ func (bc *Blockchain) canCommitGrandparent(currentBlock *BlockNode) (_grandparen
...
 
+func (bc *Blockchain) setBlockCommittedStatus(blockNode *BlockNode, status CommittedBlockStatus) {
+	bc.bestChainMap[*blockNode.Hash].CommittedStatus = status
+	bc.blockIndex[*blockNode.Hash].CommittedStatus = status
+	for _, node := range bc.bestChain {
+		if node.Hash.IsEqual(blockNode.Hash) {
+			node.CommittedStatus = status
+			break
+		}
+	}
```
Comment (tholonious) [inline lib/pos_blockchain.go:544]: I'm not sure we need to iterate over the bestChain slice here. Assuming the bestChainMap has type map[BlockHash]*BlockNode (with value being a pointer to the BlockNode), then simply updating the status of the block node in the bestChainMap should the status of the underlying BlockNode value

File: lib/pos_blockchain.go [hunk 2]
```diff
@@ -530,6 +533,17 @@ func (bc *Blockchain) canCommitGrandparent(currentBlock *BlockNode) (_grandparen
 	return nil, false
 }
 
+func (bc *Blockchain) setBlockCommittedStatus(blockNode *BlockNode, status CommittedBlockStatus) {
```
Comment (tholonious) [inline lib/pos_blockchain.go:536]: Nit: pass in the minimum info needed to perform the work in a function. In this case, the function only needs the block hash
```suggestion
func (bc *Blockchain) setBlockCommittedStatus(blockHash *BlockHash, status CommittedBlockStatus) {
```

File: lib/pos_blockchain.go (hunk 1 above)
Comment (lazynina) [reply lib/pos_blockchain.go:544]: Yep - it *should* be using the same pointer.

**Review by AeonSw4n — APPROVED** [review]
Looking good man!

File: lib/block_view.go [hunk 3]
```diff
@@ -4400,6 +4400,11 @@ func (bav *UtxoView) GetSpendableDeSoBalanceNanosForPublicKey(pkBytes []byte,
 	numImmatureBlocks := uint32(bav.Params.BlockRewardMaturity / bav.Params.TimeBetweenBlocks)
 	immatureBlockRewards := uint64(0)
 
+	// If we are passed the PoS cutover, we no longer check for immature block rewards.
+	if tipHeight >= bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight {
+		return bav.GetDeSoBalanceNanosForPublicKey(pkBytes)
+	}
```
[the remark this answers is not in the document]
Comment (lazynina) [reply lib/block_view.go:4406]: We should no longer have block rewards once we move to PoS, right? Also, we can't retrieve block rewards from the DB if a block hasn't been committed. This change is really introduced to get tests to work as all transactions will fail as we are unable to look up block rewards for uncommitted blocks. If there's an alternative way to solve this, we can address in a separate PR.

File: lib/pos_blockchain_test.go [hunk 4]
```diff
@@ -1346,12 +1351,250 @@ func TestCanCommitGrandparent(t *testing.T) {
...
+	require.NoError(t, err)
+	// Okay so let's make sure the block is uncommitted.
+	_verifyCommitRuleHelper(testMeta, []*BlockHash{}, []*BlockHash{blockHash1}, nil)
+
+	// Add one more block to best chain. Should still not trigger commit rule
+	blockTemplate2 := _generateBlockAndAddToBestChain(testMeta, 12, 12, 813)
+
+	// Run commit rule again. Nothing should happen.
+	// We expect both block 1 and block 2 to be uncommitted.
+	err = testMeta.chain.runCommitRuleOnBestChain()
```
[the remark this answers is not in the document]
Comment (lazynina) [reply lib/pos_blockchain_test.go:1374]: sure that seems fine. Can adjust in a separate PR as that code is not touch in this PR.

[the remark this answers is not in the document]
Comment (lazynina) [reply lib/pos_blockchain_test.go:1374]: Addressed in #750

File: lib/block_view.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (lazynina) [reply lib/block_view.go:4404]: abandoning this PR in favor of #761 - there you'll see that I don't touch the get spendable deso balance function.

**Comment by lazynina** [comment]
Closed in favor of #761

**Pull request closed without merging**